mobile/lib/presentation/widgets/asset_viewer/asset_viewer.page.dart

```diff
@@ -612,6 +615,15 @@ class _AssetViewerState extends ConsumerState<AssetViewer> {
       });
     });
 
+    // Listen for control visibility changes and change system UI mode accordingly
+    ref.listen(assetViewerProvider.select((value) => value.showingControls), (_, showingControls) async {
```

Comment:
can we just not tap onto the onscalestatechanged & see if it's not initial then always show the immersiveSticky ?